### PR TITLE
Create independent Snakemake language

### PR DESCRIPTION
**Pull request by SilasK**
## Description

Snakemake language was included under python. this PR makes it an independent language under the parent language python.

Colour `#419179` taken from the S in the Snakemake logo on https://snakemake.github.io/

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      
      -  `.smk` https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Asmk+NOT+nothack
      - `Snakefile`: https://github.com/search?q=filename%3ASnakefile+NOT+nothack&type=Code
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - [Tutorial](https://github.com/snakemake/snakemake/blob/1c9ea65d59d08c8c9660ea05727227ec03768383/docs/tutorial/advanced.rst)
      - See previous.
    - Sample license(s): MIT
 - [x] I created a unique ID
  - [x] I have included a syntax highlighting grammar: 
  I use python as before
  - [ ] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

**Comment by SilasK** [comment]
This is a folow up from https://github.com/github/linguist/pull/3953 and should interest @johanneskoester and maybe @nh13

**Comment by SilasK** [comment]
I removed the .smk extension from python. In this case there is no other language using this extension. isn't it?

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -6411,6 +6409,21 @@ Smithy:
...
+  codemirror_mode: python
+  codemirror_mime_type: text/x-python
+  color: "#33c68a"
+  extensions:
+  - ".smk"
+  - ".snakefile"
+  filenames:
+  - Snakefile
+  aliases:
+  - snakefile
```
Comment (lildude) [inline lib/linguist/languages.yml:6426]: We need a `language_id`. See [point 4](https://github.com/github/linguist/blob/master/CONTRIBUTING.md#adding-a-language)

**Review by lildude — CHANGES_REQUESTED** [review]
Please add a `.snakefile` sample too.

**Comment by SilasK** [comment]
@johanneskoester is it ok to set [](https://g.co/kgs/r8SHNw) as the snakemake color?

**Comment by lildude** [comment]
> @johanneskoester is it ok to set https://g.co/kgs/r8SHNw as the snakemake color?

Maybe go for `#419179` (vs your proposal of `#33c68a`) as it is the colour of the S in the logo on https://snakemake.github.io/ so it's a colour that can be traced to some sort of "branding"?

**Comment by SilasK** [comment]
@lildude I changed the color.

**Comment by SilasK** [comment]
@lildude I think I made all requested changes, didn't I ?
> Please add a `.snakefile` sample too.

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -6501,6 +6499,22 @@ Smithy:
...
+  tm_scope: source.python
+  ace_mode: python
+  codemirror_mode: python
+  codemirror_mime_type: text/x-python
+  color: "#419179"
+  extensions:
+  - ".smk"
+  - ".snakefile"
+  filenames:
+  - Snakefile
```
Comment (lildude) [inline lib/linguist/languages.yml:6514]: I've just remembered that we're going to need a sample file of this name at `samples/Snakemake/filenames/Snakefile` too. I thought we had a test that would catch this, but seems not.

Comment (SilasK) [reply lib/linguist/languages.yml:6514]: done

**Review by lildude — APPROVED** [review]

**Comment by SilasK** [comment]
💯  Thank you @lildude

**Comment by corneliusroemer** [comment]
Amazing stuff! I independently had that idea and opened an issue for it https://github.com/snakemake/snakemake/issues/2153 but that was unnecessary :)

**Pull request merged**